profiles/lib/handlers.lua

```diff
@@ -399,15 +399,22 @@ function Handlers.handle_oneway(way,result,data,profile)
   end
 end
 
+function Handlers.handle_weights(way,result,data,profile)
+  if properties.weight_name == 'distance' then
+    -- set weight rates to 1 for the distance weight, edge weights are distance / rate
+    result.forward_rate = 1
```

Comment:
Good point!  But if `{forward,backward}_mode != mode.inaccessible` then assertion `parsed_way.duration > 0 || parsed_way.forward_speed > 0` in `ProcessWay` will fail if speed is 0,
so the condition "`{forward,backward}_mode != mode.inaccessible` and `{foward,backward}_speed > 0`" is asserted to be true and if `{forward,backward}_mode == mode.inaccessible`  the weight or rate is not used to compute the edge weight.

What i missed is an additional assertion in `ProcessWay` if `parsed_way.forward_travel_mode != TRAVEL_MODE_INACCESSIBLE` and `!fallback_to_duration` then must be asserted `parsed_way.weight > 0 || parsed_way.forward_rate > 0`. At the moment such edges will have default initialized weights that is 0 for `ValueByEdge` and `ValueByMeter`.